conan/api/subapi/remotes.py

```diff
@@ -48,42 +48,39 @@ def disable(self, pattern):
         remotes = self.list(pattern, only_enabled=False)
         for r in remotes:
             r.disabled = True
-            self.update(r)
+            self.update(r.name, disabled=True)
         return remotes
 
     def enable(self, pattern):
         remotes = self.list(pattern, only_enabled=False)
         for r in remotes:
             r.disabled = False
-            self.update(r)
+            self.update(r.name, disabled=False)
         return remotes
 
     def get(self, remote_name):
         app = ConanApp(self.conan_api.cache_folder)
         return app.cache.remotes_registry.read(remote_name)
 
-    def add(self, remote: Remote, force=False):
+    def add(self, remote: Remote, force=False, index=None):
         app = ConanApp(self.conan_api.cache_folder)
-        app.cache.remotes_registry.add(remote, force=force)
+        app.cache.remotes_registry.add(remote, force=force, index=index)
 
-    def remove(self, remote_name):
+    def remove(self, pattern: str):
         app = ConanApp(self.conan_api.cache_folder)
-        remotes = self.list(remote_name, only_enabled=False)
+        remotes = self.list(pattern, only_enabled=False)
         for remote in remotes:
             app.cache.remotes_registry.remove(remote.name)
             users_clean(app.cache.localdb, remote.url)
 
-    def update(self, remote: Remote):
-        app = ConanApp(self.conan_api.cache_folder)
-        app.cache.remotes_registry.update(remote)
-
-    def move(self, remote: Remote, index: int):
+    def update(self, remote_name, url=None, secure=None, disabled=None, index=None):
         app = ConanApp(self.conan_api.cache_folder)
-        app.cache.remotes_registry.move(remote, index)
+        app.cache.remotes_registry.update(remote_name, url, secure, disabled=disabled,
+                                          index=index)
 
-    def rename(self, remote: Remote, new_name: str):
+    def rename(self, remote_name: str, new_name: str):
```

Comment:
Same, this API was changed to only require the previous name and the new one, not a full ``Remote`` object to just change the name